analytics/tests/test_counts.py

```diff
@@ -1099,6 +1138,62 @@ def assertInviteCountEquals(count: int) -> None:
         do_resend_user_invite_email(PreregistrationUser.objects.first())
         assertInviteCountEquals(6)
 
+    def test_messages_read_hour(self) -> None:
+        property = 'messages_read::hour'
+
+        user1 = self.create_user()
+        user2 = self.create_user()
+        stream, recipient = self.create_stream_with_recipient()
+        self.subscribe(user1, recipient)
+        self.subscribe(user2, recipient)
+
+        self.send_personal_message(user1, user2)
+        client = get_client("website")
+        do_mark_all_as_read(user2, client)
+        self.assertEqual(1, UserCount.objects.filter(property=property)
+                         .aggregate(Sum('value'))['value__sum'])
+
+        self.send_stream_message(user1, stream.name)
+        self.send_stream_message(user1, stream.name)
+        do_mark_stream_messages_as_read(user2, client, stream)
+        self.assertEqual(3, UserCount.objects.filter(property=property)
+                         .aggregate(Sum('value'))['value__sum'])
+
+        message = self.send_stream_message(user2, stream.name)
+        do_update_message_flags(user1, client, 'add', 'read', [message])
+        self.assertEqual(4, UserCount.objects.filter(property=property)
+                         .aggregate(Sum('value'))['value__sum'])
+
+        # TODO: Add tests where do_update_pointer is called.
```

Comment:
Not sure how to test when do_update_pointer is called.

Reply:
I think this might be a case where the right plan is just to merge our upcoming PR that will delete `do_update_pointer`.